WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -29,75 +29,114 @@ class RegisterDomainCoordinator {
-        guard let domain else { return }
-        let siteID = site?.dotComID?.intValue
-        let proxy = RegisterDomainDetailsServiceProxy()
-        proxy.createPersistentDomainShoppingCart(siteID: siteID,
-                                                 domainSuggestion: domain.remoteSuggestion(),
-                                                 privacyProtectionEnabled: domain.supportsPrivacy ?? false,
-                                                 success: { _ in
-            onSuccess()
-        },
-                                                 failure: { _ in
-            onFailure()
-        })
-    }
-
-    func presentWebViewForNoSite(on viewController: UIViewController) {
-        guard let domain,
-              let url = URL(string: Constants.noSiteCheckoutWebAddress) else {
-            crashLogger.logMessage("Failed to present domain checkout webview for no-site.",
-                                   level: .error)
-            return
-        }
-
-        presentCheckoutWebview(on: viewController,
-                               domainSuggestion: domain,
-                               url: url,
-                               title: TextContent.checkoutTitle,
-                               shouldPush: true)
+    /// Adds the selected domain to the cart then launches the checkout webview.
+    /// This flow support purchasing domains only, without plans.
+    func handlePurchaseDomainOnly(on viewController: UIViewController,
+                                  onSuccess: @escaping () -> (),
+                                  onFailure: @escaping () -> ()) {
+        createCart(
+            onSuccess: { [weak self] in
+                guard let self else { return }
+                self.presentCheckoutWebview(on: viewController, title: nil, shouldPush: false)
+                onSuccess()
+            },
+            onFailure: onFailure
+        )
     }
 
-    func presentWebViewForCurrentSite(on viewController: UIViewController) {
-        guard let domain,
-              let site,
-              let homeURL = site.homeURL,
-              let siteUrl = URL(string: homeURL as String), let host = siteUrl.host,
-              let url = URL(string: Constants.checkoutWebAddress + host) else {
-            crashLogger.logMessage("Failed to present domain checkout webview for current site.",
-                                   level: .error)
-            return
-        }
+    /// Adds the selected domain to the cart then executes `domainAddedToCartCallback` if set.
+    func addDomainToCart(on viewController: UIViewController,
+                         onSuccess: @escaping () -> (),
+                         onFailure: @escaping () -> ()) {
+        createCart(
+            onSuccess: { [weak self] in
+                guard let self = self,
+                      let domain = self.domain,
+                      let blog = site else {
+                    return
+                }
```

Comment:
❓ In `createCart` implementation, a domain can be added to a cart even if `self.site` is `nil`, but here we are checking if `site` is not `nil`. I find that odd, don't you think? 🤔

Reply:
> Technically the domainAddedToCartCallback should be called because createCart was successful

Updated the callback and function name to offer more clarity. Check 577f2453d47f80ccddb1dbb24e8e72935971d5bb

Reply:
> Changing the createCart signature to the following can resolve this issue

Addressed in caa3bb7e2127613bea0e9ba624e295066d540dec 👍